tests/sample/ext/drv/sample_ext.c

```diff
@@ -15,65 +15,85 @@
 
 // f788ef4a-207d-4dc3-85cf-0f2ea107213c
-DEFINE_GUID(EBPF_PROGRAM_TYPE_TEST, 0xf788ef4a, 0x207d, 0x4dc3, 0x85, 0xcf, 0x0f, 0x2e, 0xa1, 0x07, 0x21, 0x3c);
+DEFINE_GUID(EBPF_PROGRAM_TYPE_SAMPLE, 0xf788ef4a, 0x207d, 0x4dc3, 0x85, 0xcf, 0x0f, 0x2e, 0xa1, 0x07, 0x21, 0x3c);
 
-static ebpf_context_descriptor_t _test_ebpf_context_descriptor = {
-    sizeof(test_program_context_t),
-    EBPF_OFFSET_OF(test_program_context_t, data_start),
-    EBPF_OFFSET_OF(test_program_context_t, data_end),
-    -1};
+static ebpf_context_descriptor_t _sample_ebpf_context_descriptor = {sizeof(test_program_context_t),
+                                                                    EBPF_OFFSET_OF(test_program_context_t, data_start),
+                                                                    EBPF_OFFSET_OF(test_program_context_t, data_end),
+                                                                    -1};
 
 // Test Extension Helper function prototype descriptors.
-static ebpf_helper_function_prototype_t _test_ebpf_extension_helper_function_prototype[] = {
+static ebpf_helper_function_prototype_t _sample_ebpf_extension_helper_function_prototype[] = {
     {EBPF_MAX_GENERAL_HELPER_FUNCTION + 1,
-     "test_ebpf_extension_helper_function1",
+     "sample_ebpf_extension_helper_function1",
      EBPF_RETURN_TYPE_INTEGER,
      {EBPF_ARGUMENT_TYPE_PTR_TO_CTX}},
     {EBPF_MAX_GENERAL_HELPER_FUNCTION + 2,
-     "test_ebpf_extension_helper_function2",
+     "sample_ebpf_extension_helper_function2",
      EBPF_RETURN_TYPE_VOID,
      {EBPF_ARGUMENT_TYPE_PTR_TO_MEM, EBPF_ARGUMENT_TYPE_CONST_SIZE}},
     {EBPF_MAX_GENERAL_HELPER_FUNCTION + 3,
-     "test_ebpf_extension_helper_function3",
+     "sample_ebpf_extension_helper_function3",
      EBPF_RETURN_TYPE_VOID,
-     {EBPF_ARGUMENT_TYPE_ANYTHING}}};
+     {EBPF_ARGUMENT_TYPE_ANYTHING}},
+    {EBPF_MAX_GENERAL_HELPER_FUNCTION + 4,
+     "sample_ebpf_extension_find",
+     EBPF_RETURN_TYPE_INTEGER,
+     {EBPF_ARGUMENT_TYPE_PTR_TO_MEM,
+      EBPF_ARGUMENT_TYPE_CONST_SIZE,
+      EBPF_ARGUMENT_TYPE_PTR_TO_MEM,
+      EBPF_ARGUMENT_TYPE_CONST_SIZE}},
+    {EBPF_MAX_GENERAL_HELPER_FUNCTION + 5,
+     "sample_ebpf_extension_replace",
+     EBPF_RETURN_TYPE_INTEGER,
+     {EBPF_ARGUMENT_TYPE_PTR_TO_MEM,
+      EBPF_ARGUMENT_TYPE_CONST_SIZE,
+      EBPF_ARGUMENT_TYPE_ANYTHING,
+      EBPF_ARGUMENT_TYPE_PTR_TO_MEM,
+      EBPF_ARGUMENT_TYPE_CONST_SIZE}}};
 
-static ebpf_program_info_t _test_ebpf_extension_program_info = {
-    {"test", &_test_ebpf_context_descriptor, {0}},
-    EBPF_COUNT_OF(_test_ebpf_extension_helper_function_prototype),
-    _test_ebpf_extension_helper_function_prototype};
+static ebpf_program_info_t _sample_ebpf_extension_program_info = {
+    {"sample", &_sample_ebpf_context_descriptor, {0}},
+    EBPF_COUNT_OF(_sample_ebpf_extension_helper_function_prototype),
+    _sample_ebpf_extension_helper_function_prototype};
 
 // Test Extension helper function addresses table.
 static int
```

Comment:
typically helper functions return int64_t with a negative value meaning error

Reply:
Done. I have also removed function2 and function3 as the find/replace functions were testing the same argument types that I wanted to test.